### ISSUE TITLE
Errors are not shown

### ISSUE THREAD

**Issue by thatbakamono** [chalet-org/chalet#162]
I tried using `glm` as an external dependency in my project, it seems like CMake is doing fine, it properly detects everything but then something fails during build and the error is not really there.
```
»  Build: glm

-- The CXX compiler identification is MSVC 19.32.31329.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.32.31326/bin/HostX64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- GLM: Version 0.9.9.9
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/Bakamono/Desktop/ChaletSandbox/build/x64-pc-windows-msvc17_Release/chalet_external/glm
GLM: Visual C++ - MSVC compiler
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_C_COMPILER




X  Failed!
   Review the errors above.
```

**Comment by rewrking** [comment]
Hmm, that one is a little odd. It looks like configure step maybe returned non-zero for the warning. I'll look into it though. If you open `.chaletrc`, what are the paths to `compilerC` and `compilerCpp` in the `x64-pc-windows-msvc17` toolchain?

**Comment by thatbakamono** [comment]
```
			"compilerCpp": "C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.32.31326/bin/HostX64/x64/cl.exe",
			"compilerC": "C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.32.31326/bin/HostX64/x64/cl.exe",
```

**Comment by thatbakamono** [comment]
`fmt` and `glfw` also emitted similar warnings:
```
»  Build: fmt

-- CMake version: 3.23.1
-- The CXX compiler identification is MSVC 19.32.31329.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.32.31326/bin/HostX64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Module support is disabled.
-- Version: 8.1.2
-- Build type: Release
-- CXX_STANDARD: 11
-- Required features: cxx_variadic_templates
-- Target 'doc' disabled (requires doxygen)
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - not found
-- Found Threads: TRUE
-- FMT_PEDANTIC: OFF
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/Bakamono/Desktop/ChaletSandbox/build/x64-pc-windows-msvc17_Release/chalet_external/fmt
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_C_COMPILER
```
```
»  Build: glfw

-- The C compiler identification is MSVC 19.32.31329.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.32.31326/bin/HostX64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE)
-- Including Win32 support
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/Bakamono/Desktop/ChaletSandbox/build/x64-pc-windows-msvc17_Release/chalet_external/glfw
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_CXX_COMPILER
```
and they built just fine so it's probably not related to warnings, I would guess ninja errored out but chalet didn't print the error/overwrote it.

**Comment by rewrking** [comment]
Okay, so what's happening is glm has a CMake target name of `glm\all` and I'm assuming you probably have just `glm` defined. CMake doesn't find the target, provides no feedback for this, and exits with non-zero.

The workaround is to use:

`"targets": "glm\\all"`

A couple things I can do here:

- This might be glm/all on non-windows and glm\\all on windows, so I'll make this handle the path separator so you can just use `glm/all`
- It definitely needs to print some kind of error. I'll look into if there's a way to query cmake for the list of targets.

Edit: The same thing is probably happening with the other two. I only use `FMT_HEADER_ONLY` with fmt, but I will check them.

**Comment by thatbakamono** [comment]
I redirected the output of `chalet build` to a file and what's interesting, the error has shown:
```
C:\Users\Bakamono\Desktop\ChaletSandbox>chalet build > output.log
ninja: error: unknown target 'glm', did you mean 'all'?
```

**Comment by rewrking** [comment]
Thanks. I'll look into that too.

**Comment by rewrking** [comment]
A subtle bug I noticed with glm is that if you don't specify `BUILD_SHARED_LIBS=ON` or `BUILD_STATIC_LIBS=ON`: the first time CMake's generator step runs, `glm/all` is created as a `phony` target, but generates the target in the ninja file correctly on the second run. This stems from their CMakeLists.txt, but it's maybe a CMake bug w/ how they generate their ninja files and set defaults for those two variables.

**Comment by rewrking** [comment]
Fixed in [v0.4.2](https://github.com/chalet-org/chalet/releases/tag/v0.4.2)
